dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java

```diff
@@ -1045,56 +1066,263 @@ public void testSaveFieldVariable_KeyWithSpecialChars_ShouldSucceed(final String
+            final GraphQLFieldNameCompatibilityTestCase testCase)
             throws DotSecurityException, DotDataException {
 
         final ContentType type = new ContentTypeDataGen().nextPersisted();
         try {
-            Field field1 = FieldBuilder.builder(TextField.class)
-                    .name(fieldName)
-                    .contentTypeId(type.id())
-                    .indexed(false)
-                    .listed(false)
-                    .fixed(true)
-                    .build();
-            field1 = fieldAPI.save(field1, user);
-
-            Assert.assertNotNull(field1);
-            Assert.assertTrue(UtilMethods.isSet(field1.variable()));
-            Assert.assertNotEquals(fieldName, field1.variable());
-
-            // let's create a new field to make sure it's getting a new variable
-
-            Field field2 = FieldBuilder.builder(TextField.class)
-                    .name(fieldName)
-                    .contentTypeId(type.id())
-                    .indexed(false)
-                    .listed(false)
-                    .fixed(true)
-                    .build();
-            field2 = fieldAPI.save(field2, user);
-
-            Assert.assertNotNull(field2);
-            Assert.assertTrue(UtilMethods.isSet(field2.variable()));
-            Assert.assertNotEquals(fieldName, field2.variable());
-
-            // let's compare the two field vars are different
+            final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+            final Field field = createField(contentType, testCase.fieldName,
+                    testCase.fieldType);
+
+            if(testCase.shouldCreateNewVariable) {
+                Assert.assertNotEquals(testCase.fieldName, field.variable());
+            } else  {
+                Assert.assertEquals(testCase.fieldName, field.variable());
+            }
+        } finally {
+            contentTypeAPI.delete(type);
+        }
+    }
 
-            Assert.assertNotEquals(field1.variable(), field2.variable());
+    static class GraphQLFieldNameCompatibilityTestCase {
+        String fieldName;
+        private Class<? extends Field> fieldType;
+        boolean shouldCreateNewVariable = true;
+        String testCaseName;
 
+        @Override
+        public String toString() {
+            return testCaseName;
+        }
+    }
 
-        } finally {
-            contentTypeAPI.delete(type);
+    public static Field createField(final ContentType contentType, final String fieldName,
+            final Class<? extends Field> fieldType) {
+        try {
+            final FieldAPI fieldAPI = APILocator.getContentTypeFieldAPI();
+            final FieldBuilder fieldBuilder = getFieldBuilder(fieldType);
+            final Field field =  fieldBuilder.contentTypeId(contentType.id())
+                    .name(fieldName).build();
+            return fieldAPI.save(field, APILocator.systemUser());
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995822)